pkg/authorization/authorizer/authorizer.go

```diff
@@ -29,15 +29,15 @@ func NewSubjectLocator(delegate authorizerrbac.SubjectLocator) SubjectLocator {
 
 func (a *openshiftAuthorizer) Authorize(attributes authorizer.Attributes) (authorizer.Decision, string, error) {
 	if attributes.GetUser() == nil {
-		return authorizer.DecisionDeny, "", errors.New("no user available on context")
+		return authorizer.DecisionNoOpinion, "", errors.New("no user available on context")
 	}
 	authorized, delegateReason, err := a.delegate.Authorize(attributes)
 	if authorized == authorizer.DecisionAllow {
 		return authorizer.DecisionAllow, reason(attributes), nil
 	}
 	// errors are allowed to occur
 	if err != nil {
-		return authorizer.DecisionDeny, "", err
+		return authorized, "", err
```

Comment:
Given "authorized" seem to be able to carry either allow or deny decisions, can we make this PR also change the name to something like authorizationDecision ?
It is quite confused to see "authorized" returned when it carries a deny decision, and makes it more dificult to reason about authz flows.

Reply:
agree